ADR/####-slsa-source-provenance-verification.md

```diff
@@ -0,0 +1,130 @@
+3. **Chained attestations**: Create build provenance attestations about source provenance attestations
+
+The challenge is that the source track's storage model (git notes in repositories) differs significantly from our build artifact storage model (OCI registries), making it non-obvious how to integrate source provenance verification into our existing Conforma policies and attestation chains.
+
+### References
+
+- [SLSA Source Track Requirements](https://slsa.dev/spec/draft/source-requirements) - Source provenance specification
+- [SLSA Verification Summary Attestations](https://slsa.dev/spec/draft/verification_summary) - VSA format specification
+- [ADR 48: Movable Build Tests](0048-movable-build-tests.md) - Pattern for independent verification tasks
+- [ADR 49: VSA Support](0049-vsa-support.md) - Trusting VSAs for externally-built artifacts
+- [ADR 36: Trusted Artifacts](0036-trusted-artifacts.md) - OCI storage patterns for artifacts
+- [ADR 20: Source Retention](0020-source-retention.md) - Existing source requirements
+
+## Decision
+
+We will verify SLSA source provenance through a **chained attestation** approach that leverages the movable test pattern from [ADR 48](0048-movable-build-tests.md).
+
+### Verification Workflow
+
+1. **Source VSA generation** (external to Konflux):
+   - Source repository maintainers configure automation (GitHub Actions, GitLab CI/CD pipelines, etc.) to generate source VSAs
+   - These VSAs are stored as git notes in the source repository, contemporaneously with commits
+   - VSAs attest to source control practices (e.g., two-person review, branch protection, audit logs)
+
+2. **Source provenance verification task** (in Konflux pipeline):
+   - A Konflux pipeline task reads the source VSA from git notes for the cloned commit
+   - The task accepts a public key parameter (provided via IntegrationTestScenario configuration) and verifies the source VSA signature against that key
+   - The task verifies the VSA contents (format, subject matching the commit, etc.)
+   - The task pushes the source VSA to the OCI registry as a separate artifact
+   - The task exposes the pullspec of the OCI representation of the VSA (digest reference) in its task results using `*_ARTIFACT_OUTPUTS` format
+
+3. **Build provenance chain** (via Tekton Chains):
+   - Tekton Chains observes the task results containing the source VSA artifact reference
+   - Chains generates and signs a SLSA build provenance attestation **about** the source provenance attestation
+   - This creates an attestation chain: `Build Provenance → Source VSA → Source Commit`
+
+4. **Verification in Conforma**:
+   - Conforma policies can verify source provenance by checking for the presence of the attestation chain
+   - Conforma trusts Tekton Chains' signatures (existing trust root)
+   - Conforma reads the public key from the test attestation's task parameters and verifies it matches the expected signer identity for the source repository
```

Comment:
This would be fetched from the provenance of that attestation?

Reply:
We cannot use the tag-based fallback as I think that has all attestations in a single manifest so Chains would not be able to generate provenance for it.

But I think the last point is what I was saying as well. Conforma would need to be able to understand that this is a source VSA attestation and then verify that the keyid is the same as what is required from that source VSA attestation's provenance.